src/Popover/components/ContentWrapper.js

```diff
@@ -0,0 +1,272 @@
+      const popoverDimensions = this.popover.current.getBoundingClientRect();
+
+      // container positions and dimensions for calculation
+      const containerTop = containerDimensions.top;
+      const containerLeft = containerDimensions.left;
+      const containerHeight = containerDimensions.height;
+      const containerWidth = containerDimensions.width;
+
+      // popover dimensions for calculation
+      const popoverHeight = popoverDimensions.height;
+      const popoverWidth = popoverDimensions.width;
+
+      // window dimensions for calculation
+      const windowWidth = window.innerWidth;
+      const windowHeight = window.innerHeight;
+
+      this.containerTop = containerTop;
+      this.containerLeft = containerLeft;
+      this.containerHeight = containerHeight;
+      this.containerWidth = containerWidth;
+      this.popoverHeight = popoverHeight;
+      this.popoverWidth = popoverWidth;
+      this.windowWidth = windowWidth;
+      this.windowHeight = windowHeight;
+
+      this.contentHeight =
+        this.content.current && this.content.current.getBoundingClientRect().height;
+    }
+  }
+
+  setPosition(desiredPositions: Positions[], desiredAnchor: Anchors[]) {
+    const {
+      containerTop,
+      containerLeft,
+      containerWidth,
+      containerHeight,
+      popoverHeight,
+      popoverWidth,
+      windowHeight,
+      windowWidth,
+    } = this;
+
+    const canBePositionTop = containerTop - popoverHeight > 0;
+    const canBePositionBottom = containerTop + containerHeight + popoverHeight < windowHeight;
+
+    const canBeAnchorLeft = containerLeft + popoverWidth < windowWidth;
+    const canBeAnchorRight = containerLeft + containerWidth >= popoverWidth;
+    // returns the position name if the position can be set
+    const isInside = (p: Positions) => {
+      if (p === POSITIONS.TOP && canBePositionTop) {
+        return POSITIONS.TOP;
+      } else if (p === POSITIONS.BOTTOM && canBePositionBottom) {
+        return POSITIONS.BOTTOM;
+      }
+      return false;
+    };
+
+    const isInsideAnchor = (p: Anchors) => {
+      if (p === ANCHORS.START && canBeAnchorLeft) {
+        return ANCHORS.START;
+      } else if (p === ANCHORS.END && canBeAnchorRight) {
+        return ANCHORS.END;
+      }
+      return false;
+    };
+
+    const possiblePositions = desiredPositions
+      .map(p => isInside(p))
+      // filter all non string values
+      .filter(p => typeof p === "string");
+
+    const possibleAnchor = desiredAnchor
+      .map(p => isInsideAnchor(p))
+      .filter(p => typeof p === "string");
+
+    // set the first valid position
+    // ordering in POSITIONS const is important
+    const position = possiblePositions[0];
+    if (typeof position === "string") {
+      this.setState({ position });
+    }
+
+    const anchor = possibleAnchor[0];
+    if (typeof anchor === "string") {
+      this.setState({ anchor });
+    }
+  }
+
+  handleClickWhole = (ev: SyntheticEvent<HTMLElement>) => {
```

Comment:
`handleClose` 🙏

Reply:
I felt like it would be confusing due to handleClose prop already being there. I would prefer handleClick. It's on onClick event after all. Would that be fine?